source/base/ReferenceCounter.ooc

```diff
@@ -1,20 +1,29 @@
 import threading/Thread
 import Synchronized
 
-ReferenceCounter: class extends Synchronized {
+ReferenceCounter: class {
 	_target: Object
 	count: static Int = 0
 	_count: Int = 0
 	_kill := false
+	_lock := Mutex new(MutexType Safe)
```

Comment:
Why not. I changed it to accept the `MutexType`, so now it's possible to use global mutex easily just by changing the default value in constructor.